src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -351,16 +364,42 @@ def _dep_analyzer(self):
   def _analysis_parser(self):
     return self._analysis_tools.parser
 
+  @staticmethod
+  def _compute_language_property(target, selector):
+    """Computes the a language property setting for the given target sources.
+
+    :param target The target whose language property will be calculated.
+    :param selector A function that takes a target or platform and returns the boolean value of the
+                    property for that target or platform, or None if that target or platform does
+                    not directly define the property.
+
+    If the target does not override the language property, returns true iff the property
+    is true for any of the matched languages for the target.
+    """
+    if selector(target) is not None:
+      return selector(target)
+
+    prop = False
+    if target.has_sources('.java'):
+      prop |= selector(Java.global_instance())
+    if target.has_sources('.scala'):
+      prop |= selector(ScalaPlatform.global_instance())
+    return prop
+
   def _fingerprint_strategy(self, classpath_products):
-    return ResolvedJarAwareTaskIdentityFingerprintStrategy(self, classpath_products)
+    return ResolvedJarAwareTaskIdentityFingerprintStrategy(self, classpath_products, self._dep_context)
+
+  @staticmethod
+  def enable_strict_deps(target):
```

Comment:
I think it'd be clearer if this were named `strict_deps_enabled`. It sounds more like a setter with the current name.